catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java

```diff
@@ -188,11 +189,11 @@ public final Attribute getAttribute(String name) {
         if (Metacard.ID.equals(name)) {
             return this.id;
         }
-        for (Attribute attribute : attributes) {
-            if (attribute == null || StringUtils.isEmpty(attribute.getName())) {
-                continue;
-            } else if (name.equals(attribute.getName())) {
-                return attribute;
+        for (Attribute attribute : Collections.unmodifiableList(attributes)) {
```

Comment:
Joe is, of course, correct. The change from yesterday is still not sufficient.

Wrapping with `Collections.unmodifiableList` isn't going to be enough. Making a defensive copy in the loop, while a more expensive option, is a safer solution.

```java
for (Attribute attribute : new ArrayList<>(attributes)) {
```